

TebLocalPlannerROS:

 odom_topic: /odom
 map_frame: /odom
 
 cmd_angle_instead_rotvel: True
    
 # Trajectory
  
 teb_autosize: True
 dt_ref: 0.3
 dt_hysteresis: 0.1
 global_plan_overwrite_orientation: True
 max_global_plan_lookahead_dist: -1 # Setting to negative disables this restriction. It is limited by the local cost map settings
 feasibility_check_no_poses: 5
    
 # Robot     
 max_vel_x: 10 # 26.82 
 max_vel_x_backwards: 10.0
 max_vel_theta: 1
 acc_lim_x: 10
 acc_lim_theta: 1
 min_turning_radius: 1.0
 wheel_base: 0.57
 footprint_model: # types: "point", "circular", "two_circles", "line", "polygon"
   # vertices: [[0.125,0.45], [0.125,-0.45], [-0.125,-0.45], [-0.125,0.45]] # for type "polygon"
   type: "point"

 # GoalTolerance
    
 xy_goal_tolerance: 1
 yaw_goal_tolerance: 0.5
 free_goal_vel: False
    
 # Obstacles
    
 min_obstacle_dist: 1.5
 include_costmap_obstacles: True
 costmap_obstacles_behind_robot_dist: 1.0
 obstacle_poses_affected: 10
 costmap_converter_plugin: ""
 costmap_converter_spin_thread: True
 costmap_converter_rate: 5

 # Optimization
    
 no_inner_iterations: 3
 no_outer_iterations: 2
 optimization_activate: True
 optimization_verbose: False
 penalty_epsilon: 0.1
 weight_max_vel_x: 700
 weight_max_vel_theta: 1
 weight_acc_lim_x: 400
 weight_acc_lim_theta: 1
 weight_kinematics_nh: 1000
 weight_kinematics_forward_drive: 1
 weight_kinematics_turning_radius: 1
 weight_optimaltime: 1
 weight_obstacle: 50
 weight_dynamic_obstacle: 10 # not in use yet
 alternative_time_cost: False # not in use yet

 # Homotopy Class Planner

 enable_homotopy_class_planning: True
 enable_multithreading: True
 simple_exploration: False
 max_number_classes: 2
 roadmap_graph_no_samples: 15
 roadmap_graph_area_width: 5
 h_signature_prescaler: 0.5
 h_signature_threshold: 0.1
 obstacle_keypoint_offset: 0.1
 obstacle_heading_threshold: 0.45
 visualize_hc_graph: False
